sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -549,3 +549,25 @@ object PreWriteCheck extends (LogicalPlan => Unit) {
     }
   }
 }
+
+/**
+ * A rule to qualify relative locations with warehouse path before it breaks in catalog
+ * operation and data reading and writing.
+ *
+ * @param catalog the session catalog
+ */
+case class QualifyLocationWithWarehouse(catalog: SessionCatalog) extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+    case c @ CreateTableV1(tableDesc, _, _) if !c.locationQualifiedOrAbsent =>
```

Comment:
Do v1 DDL commands need the same fix?

Reply:
if it's a v1 only problem, shall we qualify the location in `CreateDataSourceTableCommand`?

Reply:
Yes, it's v1-specific, but,
- if we do it in runnable commands, we have to duplicate a lot of codes which may cause potential inconsistency if we don't pay enough attention.
- there are also INSERT-like statements that have a similar problem, like what I annotated [here](https://github.com/apache/spark/pull/41733/files#diff-36c570d849f9357431c1d327f0aeaf6ad4171c51b373896e02929901dcc6724aR62). If these also need to be fixed, the cases will increase.

Reply:
Hi, @cloud-fan if you don't have any concerns to this, I'd like to get it merged today:)